src/OrchardCore.Modules/OrchardCore.Media/Processing/ImageVersionProcessor.cs

```diff
@@ -5,6 +5,9 @@
 
 namespace OrchardCore.Media.Processing
 {
+    /// <summary>
+    /// Pass through processor which allows inclusion of a version query string in the cache key.
+    /// </summary>
     public class ImageVersionProcessor : IImageWebProcessor
```

Comment:
This intrigues me. ImageSharp.Web should be able to detect if a file is new even if the name is the same and replace the cached file with a new one.

https://github.com/SixLabors/ImageSharp.Web/blob/5ced07bb255f56b62f2c819d161c17e482e72c00/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L237

What issues are you seeing?

Reply:
Hi @JimBobSquarePants 

So there's a couple of reasons for this.

- This allows us to cache bust browsers
- or cache bust cdn's

Because the version string is appended when the html is produced, rather than waiting for a request to come in (which may not happen if a cdn has it!)

We use the `ImageVersionProcessor` to force a different key for the cached image stored by ImageSharp.

The detection in ImageSharp only checks if the files lastmod date is greater than. 
I've been meaning to open an issue to discuss, but time...

```
if (cachedImageMetadata.LastWriteTimeUtc > sourceImageMetadata.LastWriteTimeUtc
     && cachedImageMetadata.LastWriteTimeUtc > DateTimeOffset.Now.AddDays(-this.options.MaxCacheDays))
```

The check didn't work for me when putting together the cache busting feature.

I know the reason properly now. 
Our sample templates (TheAgency) have 20 odd images in them, and are all created from a recipe. So the source images, even though different, with different file names, all have the same lastmod date. So anytime you swap an image, by renaming one of the other images, ImageSharp doesn't spot the difference, because the dates are the same.

The other edge case was if you replace an image with a new one, and then revert back to the old one, the check is only for greater than, not less than, so the old image remains in the cache.

Reply:
@deanmarcussen Sorry for the slow reply. 

Yep, this all makes sense, I'm actually going to use a similar approach now for an image proxy we're using in my day job.

Good point on the difference detection. I think we can fix this by splitting the `ImageMetadata` struct into a source and cache version where we add the source date to the cached version also.

That check would then become:

``` c#
// Has the cached image expired or has the source image been updated?
if (cachedImageMetadata.SourceLastWriteTimeUtc != sourceImageMetadata.LastWriteTimeUtc
    || cachedImageMetadata.LastWriteTimeUtc > DateTimeOffset.Now.AddDays(-this.options.MaxCacheDays))
```

Which would allow for any date difference. The `ImageVersionProcessor ` can handle the other edge cases then.